frankenphp.go

```diff
@@ -431,7 +439,7 @@ func updateServerContext(request *http.Request, create bool, mrh C.uintptr_t) er
 }
 
 // ServeHTTP executes a PHP script according to the given context.
-func ServeHTTP(responseWriter http.ResponseWriter, request *http.Request) error {
+func ServeHTTP(responseWriter http.ResponseWriter, request *http.Request, hooks *FrankenPHPHooks) error {
```

Comment:
I'm very open to changing this as it breaks the interface. Any suggestion is welcome.

Reply:
This feature shouldn't leak in the main library but be handled only in the Caddy module. This can be achieved by wrapping `ServeHttp` in a higher-order function (a decorator).

Reply:
Can you illustrate more about what you mean by using a higher-order function here? I'm familiar with higher-order functions, but not when it comes to Go. A simple example to illustrate the pattern in Go would be huge.

Reply:
Ah, I see. Would also fix the data-race in worker mode.